library/src/cache/patch_manager.rs

```diff
@@ -346,6 +367,18 @@ impl PatchManager {
 }
 
 impl ManagePatches for PatchManager {
+    fn on_init(&mut self) -> Result<()> {
+        // If we were booting a patch but never recorded a successful boot, we assume that
+        // the patch failed to boot. Attempt to fall back.
+        if let Some(failed_boot_patch) = self.patches_state.currently_booting_patch.clone() {
+            self.try_fall_back_from_patch(failed_boot_patch.number);
+            self.patches_state.currently_booting_patch = None;
+            self.save_patches_state()?;
```

Comment:
Should this just be the higher-level mark failed call?  Will this end up creating a failure event?

Reply:
Should this return a value so the caller can create a failure event if this cant?

Reply:
Added a TODO referencing https://github.com/shorebirdtech/updater/issues/188.